packages/browser-sdk/src/hooksManager.ts

```diff
@@ -0,0 +1,105 @@
+import { CheckEvent, RawFeatures } from "./feature/features";
+import { CompanyContext, UserContext } from "./context";
+
+/**
+ * Hook for check event.
+ */
+export type CheckHook = {
+  type: "check-is-enabled" | "check-config";
+  callback: (checkEvent: CheckEvent) => void;
+};
+
+/**
+ * Hook for company update event.
+ */
+export type CompanyHook = {
+  type: "company";
+  callback: (company: CompanyContext) => void;
+};
+
+/**
+ * Hook for user update event.
+ */
+export type UserHook = {
+  type: "user";
+  callback: (user: UserContext) => void;
+};
+
+/**
+ * Hook for features updated event.
+ */
+export type FeaturesUpdatedHook = {
+  type: "features-updated";
+  callback: (features: RawFeatures) => void;
+};
+
+type trackEvent = {
+  user: UserContext;
+  company?: CompanyContext;
+  eventName: string;
+  attributes?: Record<string, any> | null;
+};
+
+/**
+ * Hook for track event.
+ */
+export type TrackHook = {
+  type: "track";
+  callback: (trackEvent: trackEvent) => void;
+};
+
+/**
+ * Hook definition.
+ */
+export type Hook =
+  | CheckHook
+  | FeaturesUpdatedHook
+  | UserHook
+  | CompanyHook
+  | TrackHook;
+
+/**
+ * Hooks manager.
+ * @internal
+ */
+export class HooksManager {
+  private hooks: {
+    "check-config": CheckHook[];
+    "check-is-enabled": CheckHook[];
+    "features-updated": FeaturesUpdatedHook[];
+    user: UserHook[];
+    company: CompanyHook[];
+    track: TrackHook[];
+  } = {
+    "check-config": [],
+    "check-is-enabled": [],
+    "features-updated": [],
+    user: [],
+    company: [],
+    track: [],
+  };
+
+  addHook(hook: Hook): void {
+    (this.hooks[hook.type] as Hook[]).push(hook);
```

Comment:
Think it's "safer" to add the type on the field itself rather than casting here

Reply:
I meant to make this.hooks be of type Hooks[] in the definition

Reply:
updated. There's still some casting going on. I've been back and forth and it's tricky to get rid of the casts. Try pulling down the code and have a go at it if you have an idea.